modules/IntOGen.py

```diff
@@ -2,17 +2,47 @@
 """This script generates disease/target evidence based on INTOGEN data."""
 
 from __future__ import annotations
+
 import argparse
-import sys
 import logging
+import sys
+from typing import TYPE_CHECKING
 
-from pyspark.sql import DataFrame, functions as f
+from pyspark.sql import DataFrame
+from pyspark.sql import functions as f
+from pyspark.sql import types as t
 
 from common.evidence import (
-    write_evidence_strings,
     initialize_sparksession,
+    write_evidence_strings,
 )
 
+if TYPE_CHECKING:
+    from pyspark.sql import Column
+
+DOI_TO_PMID_MAPPING = {
+    "10.1038/ng.2529": "23334666",
+    "10.1038/s41588-023-01321-1": "36928603",
+    "10.1038/ng.3940": "28825729",
+    "10.1101/162784": "32025007",
+    "10.1172/JCI128227": "31483290",
+    "10.1038/s41591-019-0582-4": "31570822",
+    "10.1002/hep.27198": "24798001",
+    "10.1038/ng.3520": "26928227",
+    "10.1038/s41588-018-0078-z": "29610475",
+}
+
+
+def parse_source(source_column: Column) -> Column:
```

Comment:
Does this mean that not all of the intogen evidence is dated?

Reply:
Yes, some of the evidence (~30%) are sourced from other resources where the reference is a link to a website. I haven't tracked them down further. There are 9 of them:
```
WEB: http://www.cpgea.com/
WEB: https://ascproject.org/data-release
WEB: https://dcc.icgc.org
WEB: https://ocg.cancer.gov/programs/target/projects/osteosarcoma
WEB: https://proteomics.cancer.gov/programs/cptac
WEB: https://www.cbioportal.org/datasets
WEB: https://www.hartwigmedicalfoundation.nl/data/databank/
WEB: https://www.mbcproject.org
WEB: https://www.stjude.cloud/research-domains/cancer-survivorship/
```